plugins/readme/api_key.go

```diff
@@ -0,0 +1,87 @@
+package readme
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func APIKey() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.APIKey,
+		DocsURL:       sdk.URL("https://docs.readme.com/main/reference/intro/authentication"),
+		ManagementURL: sdk.URL("https://docs.readme.com/main/reference/intro/authentication"),
```

Comment:
The `ManagementURL` field is meant to point to the dashboard where you can create a new key, rather than the docs that explain it (that's what the `DocsURL` entry you filled above is for). I believe https://dash.readme.com/ is where you create the API key for ReadMe, is that correct?

Reply:
We'll use the management URL in future improvements. For example, we could offer users that have the 1Password extension installed, to save newly created API keys directly from within the browser when they visit the management page in the dashboard.

Reply:
So this was an intentional decision because our API key management dashboard has a few path parameters that are specific to the user's ReadMe project/version setup 😞 for example:

```
https://dash.readme.com/project/some-project/v1.0/api-key
https://dash.readme.com/project/another-project/v2.1/api-key
```

The reason I linked to the docs again is because this page contains [instructions](https://docs.readme.com/main/reference/intro/authentication#generating-new-api-keys-) on how to access the dashboard. Should I link to that section specifically?

I do love the management URL idea though, have y'all considered accepting wildcards for the management URL? Being able to pass in something like this would be super helpful for us:

```
https://dash.readme.com/project/**/v**/api-key
```

I do also have an internal ticket for setting up redirects so a user could pass in something like `https://dash.readme.com/my/api-key` and it would redirect them to the correct place in the dashboard, but don't have an ETA on that unfortunately.

Reply:
Okay, thank you for the clarification! :raised_hands:

Right now the only place we're using the management URL is in printing this message:

```Golang
helpText := fmt.Sprintf("To authenticate %s with 1Password, configure which item on 1Password to use as the %s.", pluginExecutable.Name, credentialType.Name)
helpText += fmt.Sprintf("\n\n→ If you already have your %s saved in 1Password, choose Search in 1Password.", credentialType.Name)
helpText += fmt.Sprintf("\n\n→ If you don't have your %s saved in 1Password yet, choose Import into 1Password.", credentialType.Name)

if credentialType.ManagementURL != nil {
	helpText += fmt.Sprintf("\n  You can create one here if you don't have one available:\n  %s", credentialType.ManagementURL)
}

if credentialType.DocsURL != nil {
		helpText += fmt.Sprintf("\n\n→ For more info about the %s, see the documentation:\n  %s", credentialType.Name, credentialType.DocsURL)
}
```

I propose we proceed as follows:
- I'll file an internal ticket for us to consider accepting wildcards for the management URL for when we'll use it in the browser extension.
- For now, let's omit the `ManagementURL` on the ReadMe plugin. It's an optional field. When either we have added wildcard support or when your redirect solution has shipped (whichever ships first), we can add the management URL to the ReadMe plugin. The reason I prefer to proceed with not setting the management URL for the ReadMe plugin yet is that this allows us to tweak the user experience slightly for plugins that don't have a management URL for any future uses we might have for the management URL. The fallback to the docs URL might look slightly different than using the dashboard URL.
- I'll file an internal ticket for us to slightly change the wording on that line about the DocsURL to also mention `create` so it's extra clear that that doc is the place to go for info about how to create the API Key for ReadMe (because the line above about creating a new credential is omitted when the management URL isn't set).

```suggestion
```

Reply:
Ah okay, didn't realize this field was optional. Happy with this approach! Removed in https://github.com/1Password/shell-plugins/commit/57f5f20ce0dcbdd74770c2839afb6ae7ae7d2425